moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -820,6 +820,27 @@ bool IKFastKinematicsPlugin::searchPositionIK(const geometry_msgs::Pose& ik_pose
       return false;
     }
 
+    // rotate joints by +/-360° where it is possible and useful
+    for (std::size_t i = 0; i < solutions.size(); i++)
+    {
+      for (std::size_t j = 0; j < num_joints_; ++j)
+      {
+        double distance = solutions[i][j] - ik_seed_state[j];
+        while (distance > M_PI && (!joint_has_limits_vector_[j] ||
```

Comment:
It looks like you could check the `joint_has_limits_vector_` variable at the very beginning of the second for loop instead of each while loop

Reply:
I'm not sure I interpret your edit right, but yes, if a joint is continuous, the output solution here should always be in [-pi;pi].

Reply:
Changed it so it doesn't change continuous joints.